src/http_rest_api_handler.cpp

```diff
@@ -204,24 +204,227 @@ std::string HttpRestApiHandler::preprocessInferRequest(std::string request_body)
+            return StatusCode::REST_CONTENTS_FIELD_NOT_EMPTY;
+        }
+    } else if (input->datatype() == "INT8") {
+        if (input->contents().int_contents_size() > 0) {
+            SPDLOG_ERROR("INT8 contents is not empty. Content field should be empty when using binary inputs extension.");
+            return StatusCode::REST_CONTENTS_FIELD_NOT_EMPTY;
+        }
+    } else if (input->datatype() == "UINT64") {
+        if (input->contents().uint64_contents_size() > 0) {
+            SPDLOG_ERROR("UINT64 contents is not empty. Content field should be empty when using binary inputs extension.");
+            return StatusCode::REST_CONTENTS_FIELD_NOT_EMPTY;
+        }
+    } else if (input->datatype() == "UINT32") {
+        if (input->contents().uint_contents_size() > 0) {
+            SPDLOG_ERROR("UINT32 contents is not empty. Content field should be empty when using binary inputs extension.");
+            return StatusCode::REST_CONTENTS_FIELD_NOT_EMPTY;
+        }
+    } else if (input->datatype() == "UINT16") {
+        if (input->contents().uint_contents_size() > 0) {
+            SPDLOG_ERROR("UINT16 contents is not empty. Content field should be empty when using binary inputs extension.");
+            return StatusCode::REST_CONTENTS_FIELD_NOT_EMPTY;
+        }
+    } else if (input->datatype() == "UINT8") {
+        if (input->contents().uint_contents_size() > 0) {
+            SPDLOG_ERROR("UINT8 contents is not empty. Content field should be empty when using binary inputs extension.");
+            return StatusCode::REST_CONTENTS_FIELD_NOT_EMPTY;
+        }
+    } else if (input->datatype() == "FP64") {
+        if (input->contents().fp64_contents_size() > 0) {
+            SPDLOG_ERROR("FP64 contents is not empty. Content field should be empty when using binary inputs extension.");
+            return StatusCode::REST_CONTENTS_FIELD_NOT_EMPTY;
+        }
+    } else if (input->datatype() == "BYTES") {
+        if (input->contents().bytes_contents_size() > 0) {
+            SPDLOG_ERROR("BYTES contents is not empty. Content field should be empty when using binary inputs extension.");
+            return StatusCode::REST_CONTENTS_FIELD_NOT_EMPTY;
+        }
+    } else {
+        return StatusCode::REST_UNSUPPORTED_PRECISION;
+    }
+
+    return StatusCode::OK;
+}
+
+Status addBinaryInputs(::inference::ModelInferRequest& grpc_request, const char* binary_inputs, size_t binary_inputs_size) {
+    int binary_input_offset = 0;
+    for (int i = 0; i < grpc_request.mutable_inputs()->size(); i++) {
+        auto input = grpc_request.mutable_inputs()->Mutable(i);
+        auto status = validateContentFieldsEmptiness(input);
+        if (!status.ok()) {
+            return status;
+        }
+        if (input->parameters().count("binary_data_size") > 0) {
+            if (input->parameters().at("binary_data_size").parameter_choice_case() == inference::InferParameter::ParameterChoiceCase::kInt64Param) {
```

Comment:
Using .at multiple times searches for the entry multiple times. Iterator could speed it up